File: src/main/java/doom/despair/core/events/ShipCreateEvent.java
package doom.despair.core.events;

import com.google.gson.Gson;
import doom.despair.core.*;
import doom.despair.server.Board;
import doom.despair.ships.ShipType;

public class ShipCreateEvent {
    private static class ShipCreateContext {
        private ShipType type;
    }

    public static class ServerShipCreateEvent extends ServerGameEvent {
        @Override
        public void handle(ServerContext ctx) {
            Board playerBoard = ctx.server().getBoardForPlayer(this.player);
            ShipCreateContext shipCreateContext = new Gson().fromJson(this.nested, ShipCreateContext.class);
            playerBoard.placeShip(shipCreateContext.type);
        }
    }

    public static class ClientShipCreateEvent extends ClientGameEvent {
        @Override
        public void handle(ClientContext context) {
        }
    }
}

File: src/main/java/doom/despair/core/events/ShipCreateEvent.kt
package doom.despair.core.events

import com.google.gson.*;
import doom.despair.core.ClientContext
import doom.despair.core.ClientGameEvent
import doom.despair.core.ServerContext
import doom.despair.core.ServerGameEvent
import doom.despair.server.Board
import doom.despair.ships.ShipType

class ShipCreateEvent {
    private class ShipCreateContext {
        val type: ShipType = ShipType.AIRCRAFT_CARRIER
        val start: Board.Coordinate = Board.Coordinate(0, 0)
        val horizontal = false
    }

    class ServerShipCreateEvent : ServerGameEvent() {
        override fun handle(context: ServerContext) {
            val playerBoard: Board = context.server.getBoardForPlayer(this.player)
            val scc: ShipCreateContext = Gson().fromJson(this.nested, ShipCreateContext::class.java)
            playerBoard.placeShip(scc.type, scc.start, scc.horizontal)
        }
    }

    class ClientShipCreateEvent : ClientGameEvent() {
        override fun handle(context: ClientContext) {

        }
    }
}
